### ISSUE TITLE
Semantic: container decl arguments (`enum(expr)`, `packed struct(expr)`) are never visited

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#368]
# Semantic: container decl arguments (`enum(expr)`, `packed struct(expr)`) are never visited

- **Severity:** Medium-high (default-on `unused-decls` false positive with dangerous autofix)
- **Status:** Verified by reproduction
- **Location:** `src/Semantic/Builder.zig:558-590` (`visitContainer`)

## The bug

`visitContainer` never visits `container.ast.arg` (the backing-type expression). Instead, it has a token-level special case that only handles a bare single identifier:

```zig
// Builder.zig:562
if (container.ast.enum_token == null) {
    const maybe_ident = main_token + 2;
    if (tags[main_token + 1] == .l_paren and
        tags[maybe_ident] == .identifier and
        tags[maybe_ident + 1] != .period)
    {
        ...
        _ = try self.recordReference(.{ .flags = .{ .type = true }, ... });
    }
}
```

Anything more complex than `enum(u8)` — a field access, a function call, a nested `enum(...)` inside `union(...)` — records no references, and the arg subtree's nodes never get node-links (their `parents`/`scopes` entries keep defaults).

## User impact

Declarations referenced only from a container's backing-type expression are reported unused by the default-on `unused-decls` rule, and its autofix (applied by plain `--fix` per report 04) deletes them, breaking the build. `enum(std.math.IntFittingRange(...))` and `packed struct(BackingInt)` are established idioms.

## Reproduction

```sh
cat > sem3.zig <<'EOF'
const std = @import("std");
const E = enum(std.math.IntFittingRange(0, 100)) { a };
pub fn main() void {
    _ = E.a;
}
EOF
zlint sem3.zig
# warning: unused-decls: variable 'std' is declared but never used.
```

Also affected: `union(enum(u8))` (the `enum_token != null` branch skips the arg entirely).

## Proposed fix

Delete the token-peeking special case and visit the arg node like any other type expression, inside the container's scope:

```zig
if (container.ast.arg.unwrap()) |arg| {
    try self.visitType(arg);   // records type references and node links
}
```

The existing test only covers the bare-identifier case (`symbol_ref_test.zig:538-543`); add cases for a field-access arg, a call arg, and `union(enum(u8))`.
